pkg/configauditreport/controller/controller.go

```diff
@@ -193,8 +193,7 @@ func (r *ResourceController) reconcileResource(resourceKind kube.Kind) reconcile
 		}
 
 		if r.Config.ConfigAuditScannerScanOnlyCurrentRevisions && resourceKind == kube.KindReplicaSet {
-			controller := metav1.GetControllerOf(resource)
-			activeReplicaSet, err := r.IsActiveReplicaSet(ctx, resource, controller)
+			activeReplicaSet, controller, err := r.IsActiveReplicaSet(ctx, resource.(*appsv1.ReplicaSet))
```

Comment:
why it need to be casted  ? and if you still think its  needed then it needed to be check
```suggestion
                        rsResource,ok := resource.(*appsv1.ReplicaSet)
                        if !ok {
                                       return ctrl.Result{}, fmt.Errorf("resource failed casting to replica set: %w", err)
                                    }
			activeReplicaSet, controller, err := r.IsActiveReplicaSet(ctx, rsResource)
```

Reply:
So the conditional right above is not enough (last part)? How come?

`r.Config.ConfigAuditScannerScanOnlyCurrentRevisions && resourceKind == kube.KindReplicaSet`

Reply:
I find it a bit odd to have a func named `IsActiveReplicaSet` not accepting `ReplicaSet` as param.

Reply:
still doesn't answer the question why cast it  if not needed?  also changing the method naming probably will be cheeper